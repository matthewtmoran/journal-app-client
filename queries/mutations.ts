import gql from "graphql-tag";

const SIGNIN_MUTATION = gql`
  mutation SignInMutation($email: String!, $password: String!) {
    login(data: { email: $email, password: $password }) {
      token
    }
  }
`;

const SIGNUP_MUTATION = gql`
  mutation SignupMutation($email: String!, $password: String!) {
    createUser(data: { email: $email, password: $password }) {
      token
    }
  }
`;

const CREATE_ENTRY_MUTATION = gql`
  mutation CreateEntryMutation(
    $title: String!
    $body: String
    $description: String
    $audioPath: String
    $audioFile: String
    $categories: [CreateCategoryInput!]
  ) {
    createEntry(
      data: {
        title: $title
        body: $body
        description: $description
        audioPath: $audioPath
        audioFile: $audioFile
        categories: $categories
      }
    ) {
      id
      title
      description
      createdAt
      updatedAt
      imagePath
      audioPath
      body
      categories {
        id
        name
        color
      }
    }
  }
`;

export { CREATE_ENTRY_MUTATION, SIGNIN_MUTATION, SIGNUP_MUTATION };
