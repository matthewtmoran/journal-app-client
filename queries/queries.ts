import gql from "graphql-tag";

export const ENTRIES_QUERY = gql`
  query EntiesQuery {
    entries {
      id
      title
      description
      imagePath
      audioPath
      body
      createdAt
      updatedAt
      categories {
        id
        name
        color
      }
    }
  }
`;
