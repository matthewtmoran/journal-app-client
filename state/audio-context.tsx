import React, {
  createContext,
  useMemo,
  useReducer,
  FunctionComponent,
  useContext,
  useRef,
} from "react";
import { Audio } from "expo-av";
import getMMSSFromMillis from "../utils/getMMSSFromMillis";
import padWithZero from "../utils/padWithZero";

interface IState {
  isLoading: boolean;
  isPlaybackAllowed: boolean;
  muted: boolean;
  soundPosition: null | number;
  soundDuration: number;
  shouldPlay: boolean;
  isPlaying: boolean;
  shouldCorrectPitch: boolean;
  volume: number;
  rate: number;
  playbackInstanceName: string;
}

interface IAction {
  type: string;
  status?: IStatus;
}

interface IStatus {
  isLoaded: boolean;
  didJustFinish: boolean;
  durationMillis: number;
  positionMillis: number;
  shouldPlay: boolean;
  isPlaying: boolean;
  rate: number;
  isMuted: boolean;
  volume: number;
  shouldCorrectPitch: boolean;
  error?: any;
}

const initialState: IState = {
  isLoading: false,
  isPlaybackAllowed: false,
  muted: false,
  soundPosition: null,
  soundDuration: 0,
  shouldPlay: false,
  isPlaying: false,
  shouldCorrectPitch: false,
  volume: 1,
  rate: 1,
  playbackInstanceName: "Test",
};

const UPDATE_STATUS = "UPDATE_STATUS";
const AUDIO_UNLOADED = "AUDIO_UNLOADED";
const AUDIO_LOADING = "AUDIO_LOADING";
const AUDIO_LOADING_DONE = "AUDIO_LOADING_DONE";
const AUDION_FINISHED_PLAYING = "AUDION_FINISHED_PLAYING";
const UPDATE_POSITION = "UPDATE_POSITION";

const AudioReducer = (prevState: IState, action: IAction): IState => {
  switch (action.type) {
    case AUDION_FINISHED_PLAYING: {
      return { ...prevState, soundDuration: 0, soundPosition: null };
    }
    case AUDIO_LOADING:
      return { ...prevState, isLoading: true };
    case AUDIO_UNLOADED: {
      return {
        ...prevState,
        isPlaybackAllowed: false,
        soundDuration: 0,
        soundPosition: null,
      };
    }
    case UPDATE_STATUS: {
      if (!action.status) {
        return { ...prevState };
      }
      const {
        durationMillis,
        positionMillis,
        shouldPlay,
        isPlaying,
        // isBuffering,
        rate,
        isMuted,
        volume,
        shouldCorrectPitch,
      } = action.status;
      return {
        ...prevState,
        isPlaying,
        rate,
        shouldCorrectPitch,
        shouldPlay,
        volume,
        muted: isMuted,
        soundDuration: durationMillis,
        soundPosition: positionMillis,
        isPlaybackAllowed: true,
      };
    }

    case AUDIO_LOADING_DONE: {
      return { ...prevState, isLoading: false };
    }

    case UPDATE_POSITION: {
      // @ts-ignore
      return { ...prevState, soundPosition: action.status.soundPosition };
    }

    default:
      return { ...prevState };
  }
};

const AudioProvider: FunctionComponent = ({ children }) => {
  const [state, dispatch] = useReducer(AudioReducer, initialState);
  const playbackInstance = useRef<Audio.Sound | null>(null);
  const isSeeking = useRef<boolean>(false);
  const shouldPlayAtEndOfSeek = useRef<boolean>(false);

  const updateScreenForSoundStatus = (status: IStatus) => {
    if (status.didJustFinish) {
      if (playbackInstance.current) {
        playbackInstance.current.stopAsync();
      }
      dispatch({ type: AUDION_FINISHED_PLAYING, status });
    }
    if (status.isLoaded) {
      dispatch({ type: UPDATE_STATUS, status });
    } else {
      dispatch({ type: AUDIO_UNLOADED });
    }
    if (status.error) {
      console.log(`FATAL PLAYER ERROR: ${status.error}`);
    }
  };

  const audioContext = {
    loadAudio: async (audioPath: string) => {
      dispatch({ type: AUDIO_LOADING });
      const { sound, status } = await Audio.Sound.createAsync(
        { uri: audioPath },
        {
          isLooping: false,
          isMuted: state.muted,
          volume: state.volume,
          rate: state.rate,
          shouldCorrectPitch: state.shouldCorrectPitch,
        },
        (status) => updateScreenForSoundStatus(status as any)
      );
      sound.setProgressUpdateIntervalAsync(100);
      playbackInstance.current = sound;
      dispatch({ type: AUDIO_LOADING_DONE });
    },

    unloadAudio: async () => {
      dispatch({ type: AUDIO_LOADING });
      if (playbackInstance.current) {
        playbackInstance.current.setOnPlaybackStatusUpdate(null);
        playbackInstance.current = null;
      }
      dispatch({ type: AUDIO_LOADING_DONE });
    },

    stopAudioAndUnload: async () => {
      if (playbackInstance.current) {
        await playbackInstance.current.stopAsync();
        playbackInstance.current.setOnPlaybackStatusUpdate(null);
        playbackInstance.current = null;
      }
    },

    handlePlayPausePress: () => {
      if (playbackInstance.current) {
        if (state.isPlaying) {
          playbackInstance.current.pauseAsync();
        } else {
          playbackInstance.current.playAsync();
        }
      }
    },

    onSeekSliderValueChange: (value: number) => {
      if (playbackInstance.current !== null && !isSeeking.current) {
        shouldPlayAtEndOfSeek.current = state.shouldPlay;
        playbackInstance.current.pauseAsync();
        isSeeking.current = true;
      }
    },

    onSeekSliderSlidingComplete: async (value: number) => {
      if (playbackInstance.current !== null) {
        const seekPosition = value * state.soundDuration;
        if (shouldPlayAtEndOfSeek.current) {
          await playbackInstance.current.playFromPositionAsync(seekPosition);
        } else {
          await playbackInstance.current.setPositionAsync(seekPosition);
        }
        isSeeking.current = false;
      }
    },

    getSeekSliderPosition: () => {
      if (
        playbackInstance.current != null &&
        state.soundPosition != null &&
        state.soundDuration != null
      ) {
        return state.soundPosition / state.soundDuration;
      }
      return 0;
    },

    fetMMSSFromMillis: (millis: number) => {
      const totalSeconds = millis / 1000;
      const seconds = Math.floor(totalSeconds % 60);
      const minutes = Math.floor(totalSeconds / 60);
      return padWithZero(minutes) + ":" + padWithZero(seconds);
    },

    getPlaybackTimestamp: () => {
      if (
        playbackInstance.current != null &&
        state.soundPosition != null &&
        state.soundDuration != null
      ) {
        return `${getMMSSFromMillis(state.soundPosition)} / ${getMMSSFromMillis(
          state.soundDuration
        )}`;
      }
      return "";
    },
  };

  return (
    <AudioContext.Provider value={{ ...audioContext, ...state }}>
      {children}
    </AudioContext.Provider>
  );
};

const AudioContext = createContext<any>(undefined);

const useAudio = () => {
  const context = useContext(AudioContext);
  if (context === undefined) {
    throw new Error(`useAudio must be used within a AudioProvider`);
  }
  return context;
};

export { AudioProvider, useAudio };
