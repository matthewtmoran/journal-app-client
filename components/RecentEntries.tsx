import React from "react";
import { StyleSheet, Text, View, FlatList } from "react-native";
import { useQuery } from "@apollo/react-hooks";
import EntryPreview from "./EntryPreview";
import { RobotLightText } from "./StyledText";
import { ENTRIES_QUERY } from "../queries/queries";

const RecentEntries = () => {
  const { loading, error, data } = useQuery(ENTRIES_QUERY, {});
  if (loading) {
    return (
      <View style={styles.emptyContainer}>
        <RobotLightText style={styles.message}>
          No recent entries.
        </RobotLightText>
      </View>
    );
  }
  if (error) {
    return (
      <View style={styles.emptyContainer}>
        <Text>{`Error: ${error.message}`}</Text>
      </View>
    );
  }

  const recent = data.entries
    .sort((a: any, b: any) => {
      return Number(new Date(a.createdAt)) - Number(new Date(b.createdAt));
    })
    .slice(data.entries.length - 4, data.entries.length);

  return (
    <View style={styles.container}>
      {!recent.length ? (
        <View style={styles.emptyContainer}>
          <RobotLightText style={styles.message}>
            No recent entries.
          </RobotLightText>
        </View>
      ) : (
        <FlatList
          listKey="RecentEntries"
          style={styles.entries}
          data={recent}
          keyExtractor={(item) => item.id}
          horizontal={false}
          numColumns={2}
          renderItem={({ item, index }: any) => (
            <EntryPreview entry={item} index={index} />
          )}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 22,
  },
  emptyContainer: {
    flex: 1,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  message: {
    fontStyle: "italic",
  },
  entries: {
    margin: 10,
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});

export default RecentEntries;
