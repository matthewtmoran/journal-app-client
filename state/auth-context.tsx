import React, { useReducer, createContext, useEffect, useMemo } from "react";
import { useMutation } from "@apollo/react-hooks";
import { SIGNIN_MUTATION, SIGNUP_MUTATION } from "../queries/mutations";
import { ActivityIndicator, AsyncStorage } from "react-native";
import { AUTH_TOKEN } from "../constants";
import ICredentials from "../interfaces/ICredentials";
import TransparentLoader from "../components/TransparentLoader";
import * as SplashScreen from "expo-splash-screen";

interface IAppState {
  isLoading: boolean;
  userToken: string | null;
  error: null | string;
  status:
    | typeof STATUS_REJECTED
    | typeof STATUS_RESOLVED
    | typeof STATUS_PENDING
    | typeof STATUS_IDLE
    | typeof STATUS_BOOTSTRAPPING;
}

const AuthContext = createContext<any>(undefined);

const LOGGED_OUT = "LOGGED_OUT";
const SIGN_IN = "SIGN_IN";
const SIGN_UP = "SIGN_UP";
const RESTORE_TOKEN = "RESTORE_TOKEN";
const ERROR = "ERROR";
const SUCCESS = "SUCCESS";
const STARTED = "STARTED";
const BOOTSTRAPPING = "BOOTSTRAPPING ";

export const STATUS_REJECTED = "STATUS_REJECTED";
const STATUS_RESOLVED = "STATUS_RESOLVED";
export const STATUS_PENDING = "STATUS_PENDING";
const STATUS_IDLE = "STATUS_IDLE";
const STATUS_BOOTSTRAPPING = "STATUS_BOOTSTRAPPING";

const initialState: IAppState = {
  isLoading: true,
  userToken: null,
  status: STATUS_IDLE,
  error: null,
};

interface IAction {
  type: typeof ERROR | typeof SUCCESS | typeof STARTED | typeof LOGGED_OUT;
  error?: string | null;
  token?: string | null;
}

const AuthReducer = (prevState: any, action: IAction) => {
  switch (action.type) {
    case STARTED:
      return {
        ...prevState,
        userToken: null,
        status: STATUS_PENDING,
        error: null,
      };
    case ERROR:
      return {
        ...prevState,
        userToken: null,
        status: STATUS_REJECTED,
        error: action.error,
      };
    case SUCCESS:
      return {
        ...prevState,
        userToken: action.token,
        status: STATUS_RESOLVED,
        error: null,
      };
    case LOGGED_OUT:
      return {
        ...prevState,
        userToken: null,
        status: STATUS_IDLE,
        error: null,
      };
    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

const AuthProvider: React.FunctionComponent<{ userToken: string | null }> = ({
  children,
  userToken,
}) => {
  const [state, dispatch] = useReducer(AuthReducer, initialState);

  useEffect(() => {
    const bootstrapAsync = async () => {
      if (userToken) {
        return dispatch({ type: SUCCESS, token: userToken });
      }

      // dispatch({ type: STARTED });
      // let token;
      // try {
      //   token = await AsyncStorage.getItem(AUTH_TOKEN);
      //   dispatch({ type: SUCCESS, token });
      // } catch (error) {
      //   console.log({ error });
      //   dispatch({ type: ERROR, error: error.message });
      // }
    };

    bootstrapAsync();
  }, []);

  const [login, loginEvents] = useMutation(SIGNIN_MUTATION, {
    onCompleted({ login: { token } }) {
      AsyncStorage.setItem(AUTH_TOKEN, token);
      dispatch({ type: SUCCESS, token });
    },
    onError(error) {
      console.log({ error });
      dispatch({ type: ERROR, error: error.message });
    },
  });

  const [register] = useMutation(SIGNUP_MUTATION, {
    onCompleted(data) {
      const { token } = data.createUser;
      AsyncStorage.setItem(AUTH_TOKEN, token);
      dispatch({ type: SUCCESS, token });
    },
    onError(error) {
      console.log({ error });
      dispatch({ type: ERROR, error: error.message });
    },
  });

  const authContext = useMemo(
    () => ({
      signIn: async ({ email, password }: ICredentials) => {
        dispatch({ type: STARTED });
        return login({ variables: { email, password } });
      },

      signOut: async (client: any) => {
        dispatch({ type: STARTED });
        await client.clearStore();
        await client.resetStore();
        await AsyncStorage.removeItem(AUTH_TOKEN);
        dispatch({ type: LOGGED_OUT });
      },

      signUp: async ({ email, password }: any) => {
        dispatch({ type: STARTED });
        return register({ variables: { email, password } });
      },
      reset: () => {
        dispatch({ type: LOGGED_OUT });
      },
    }),
    []
  );

  return (
    <AuthContext.Provider value={{ ...authContext, ...state }}>
      <TransparentLoader isVisible={state.status === STATUS_PENDING} />
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = () => {
  const context = React.useContext(AuthContext);
  if (context === undefined) {
    throw new Error(`useAuth must be used within a AuthProvider`);
  }
  return context;
};

export { AuthProvider, useAuth };
