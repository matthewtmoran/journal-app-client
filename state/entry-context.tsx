import React, {
  FunctionComponent,
  useContext,
  createContext,
  useReducer,
  useMemo,
} from "react";
import * as FileSystem from "expo-file-system";
import { useMutation } from "@apollo/react-hooks";
import { CATEGORIES_QUERY, ENTRIES_QUERY } from "../queries/queries";
import {
  UPDATE_ENTRY_MUTATION,
  CREATE_ENTRY_MUTATION,
  DELETE_ENTRY_MUTATION,
} from "../queries/mutations";
import {
  STATUS_REJECTED,
  STATUS_RESOLVED,
  STATUS_PENDING,
  STATUS_IDLE,
  ERROR,
  SUCCESS,
  STARTED,
} from "../constants";
import ICategory from "../interfaces/ICategory";
import IEntry from "../interfaces/IEntry";
import filterDuplicateCategories from "../utils/filterDuplicateCategories";

interface IFormValues {
  body: string;
  description: string;
  title: string;
  categories: ICategory[];
}

interface IUpdateInput extends IFormValues {
  id: string;
}

interface ICreateInput extends IFormValues {
  audioPath?: string;
}

const updateEntries = (entries: IEntry[], updated: IEntry) => {
  return entries.map((entry: IEntry) => {
    if (entry.id === updated.id) {
      return updated;
    }
    return entry;
  });
};

interface IAction {
  type: typeof ERROR | typeof SUCCESS | typeof STARTED;
  error?: string | null;
}

type StatusTypes =
  | typeof STATUS_REJECTED
  | typeof STATUS_RESOLVED
  | typeof STATUS_PENDING
  | typeof STATUS_IDLE;

interface IState {
  status: StatusTypes;
  error?: string | null;
}

const initialState: IState = {
  status: STATUS_IDLE,
  error: null,
};

const EntryReducer = (prevState: IState, action: IAction): IState => {
  switch (action.type) {
    case STARTED:
      return {
        ...prevState,
        status: STATUS_PENDING,
        error: null,
      };
    case ERROR:
      return {
        ...prevState,
        status: STATUS_REJECTED,
        error: action.error,
      };
    case SUCCESS:
      return {
        ...prevState,
        status: STATUS_RESOLVED,
        error: null,
      };
    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

const EntryProvider: FunctionComponent = ({ children }) => {
  const [state, dispatch] = useReducer(EntryReducer, initialState);
  const [updateEntry] = useMutation(UPDATE_ENTRY_MUTATION, {
    onCompleted(data) {
      dispatch({ type: SUCCESS });
    },
    onError(error) {
      dispatch({ type: ERROR, error: error.message });
    },
  });

  const [createEntry] = useMutation(CREATE_ENTRY_MUTATION, {
    onCompleted(data) {
      dispatch({ type: SUCCESS });
    },
    onError(error) {
      dispatch({ type: ERROR, error: error.message });
    },
  });

  const [deleteEntry] = useMutation(DELETE_ENTRY_MUTATION, {
    onCompleted(data) {
      dispatch({ type: SUCCESS });
    },
    onError(error) {
      dispatch({ type: ERROR, error: error.message });
    },
  });

  const entryContext = useMemo(
    () => ({
      handleCreateEntry: async ({
        title,
        body,
        description,
        categories,
        audioPath,
      }: ICreateInput) => {
        console.log("handleCreateEntry ");
        dispatch({ type: STARTED });
        const audioFile: string = await FileSystem.readAsStringAsync(
          audioPath,
          {
            encoding: FileSystem.EncodingType.Base64,
          }
        );

        createEntry({
          variables: {
            title,
            body,
            description,
            audioPath,
            audioFile,
            categories,
          },
          update: (cache, { data: { createEntry } }: any) => {
            const data: any = cache.readQuery({ query: ENTRIES_QUERY });
            const categoryData: any = cache.readQuery({
              query: CATEGORIES_QUERY,
            });

            cache.writeQuery({
              query: CATEGORIES_QUERY,
              data: {
                categories: filterDuplicateCategories(
                  categoryData.categories,
                  createEntry.categories
                ),
              },
            });

            cache.writeQuery({
              query: ENTRIES_QUERY,
              data: { entries: [...data.entries, createEntry] },
            });
          },
        });
      },

      handleUpdateEntry: async ({
        id,
        title,
        body,
        description,
        categories,
      }: IUpdateInput) => {
        dispatch({ type: STARTED });
        updateEntry({
          variables: {
            id,
            title,
            body,
            description,
            categories,
          },
          update: (cache, { data: { updateEntry } }) => {
            const { entries } = cache.readQuery<{ entries: IEntry[] }>({
              query: ENTRIES_QUERY,
            })!;
            const { categories } = cache.readQuery<{ categories: ICategory[] }>(
              {
                query: CATEGORIES_QUERY,
              }
            )!;
            cache.writeQuery({
              query: CATEGORIES_QUERY,
              data: {
                categories: filterDuplicateCategories(
                  categories,
                  updateEntry.categories
                ),
              },
            });

            cache.writeQuery({
              query: ENTRIES_QUERY,
              data: { entries: updateEntries(entries, updateEntry) },
            });
          },
        });
      },

      handleDeleteEntry: async ({ id, audioPath }: IEntry) => {
        dispatch({ type: STARTED });

        deleteEntry({
          variables: { id },
          update: (cache, { data: { deleteEntry } }) => {
            const { entries } = cache.readQuery<{ entries: IEntry[] }>({
              query: ENTRIES_QUERY,
            })!;

            cache.writeQuery({
              query: ENTRIES_QUERY,
              data: { entries: entries.filter((e) => e.id !== deleteEntry.id) },
            });
          },
        });

        try {
          FileSystem.deleteAsync(audioPath);
        } catch (error) {
          console.log(
            "There was an error deleting file from file system",
            error
          );
        }
      },
    }),
    []
  );

  return (
    <EntryContext.Provider value={{ ...entryContext, ...state }}>
      {children}
    </EntryContext.Provider>
  );
};

const EntryContext = createContext<any>(undefined);

const useEntry = () => {
  const context = useContext(EntryContext);
  if (context === undefined) {
    throw new Error(`useEntry must be used within a EntryProvider`);
  }
  return context;
};

export { EntryProvider, useEntry };
