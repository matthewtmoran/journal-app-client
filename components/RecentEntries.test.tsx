import React from "react";
import { render, wait } from "@testing-library/react-native";
import RecentEntries from "./RecentEntries";
import { MockedProvider } from "@apollo/react-testing";
import { ENTRIES_QUERY } from "../queries/queries";
import { EntryProvider } from "../state/entry-context";

describe("components/EntryPreview", () => {
  it("Matches snapshot", async () => {
    const date = new Date();
    const mocks = {
      request: {
        query: ENTRIES_QUERY,
        variables: {},
      },
      result: {
        data: {
          entries: [
            {
              __typename: "Entry",
              id: 1,
              title: "Buck",
              description: "This is a description",
              body: "this is the body",
              audioPath: "path/to/file",
              imagePath: "",
              updatedAt: date,
              createdAt: date,
              categories: [
                {
                  __typename: "Category",
                  id: 1,
                  name: "thoughts",
                  color: "#333",
                },
              ],
            },
          ],
        },
      },
    };
    const { baseElement } = render(
      <MockedProvider mocks={[mocks]} addTypename={true}>
        <EntryProvider>
          <RecentEntries />
        </EntryProvider>
      </MockedProvider>
    );
    await wait();
    expect(baseElement).toMatchSnapshot();
  });

  it("Should render loading", async () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]}>
        <EntryProvider>
          <RecentEntries />
        </EntryProvider>
      </MockedProvider>
    );
    const loadingContainer = getByTestId("entries-loading");
    expect(loadingContainer).toBeTruthy();
    await wait();
  });

  it("Should render error ui", async () => {
    const mocks = {
      request: {
        query: ENTRIES_QUERY,
        variables: {},
      },
      error: new Error("Error geting entries"),
    };

    const { getByText } = render(
      <MockedProvider mocks={[mocks]}>
        <EntryProvider>
          <RecentEntries />
        </EntryProvider>
      </MockedProvider>
    );

    await wait();
    const errorMessage = getByText(
      "Error: Network error: Error geting entries"
    );
    expect(errorMessage).toBeTruthy();
  });

  it("Should render no recent entries text", async () => {
    const date = new Date();
    const mocks = {
      request: {
        query: ENTRIES_QUERY,
        variables: {},
      },
      result: {
        data: {
          entries: [],
        },
      },
    };

    const { getByText } = render(
      <MockedProvider mocks={[mocks]} addTypename={true}>
        <EntryProvider>
          <RecentEntries />
        </EntryProvider>
      </MockedProvider>
    );

    await wait();

    const element = getByText("No recent entries.");
    expect(element).toBeDefined();
  });
});
