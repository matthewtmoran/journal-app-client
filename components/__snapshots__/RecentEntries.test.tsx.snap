// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EntryPreview Matches snapshot 1`] = `null`;
